### ISSUE TITLE
Can't batch System instructions

### ISSUE THREAD

**Issue by garious** [solana-labs/solana#1551]
SystemProgram ignores the instruction's `accounts` field and accesses the Transaction's `account_keys` field directly. That means, for example, you can't batch two independent `Move` instructions in one transaction.

**Comment by garious** [comment]
Somewhat depends on #1531. Using `accounts` would allow the same user to move tokens to multiple destinations, but adding multiple signature support would allow different users to move tokens to different destinations within the same transaction.

**Comment by sakridge** [comment]
Hmm I was just looking at messing with this stuff as part of the signatures change.

**Comment by garious** [comment]
Sounds great!

**Comment by aeyakovenko** [comment]
@garious @jackcmay i think spawn is the only culprint
```
            SystemInstruction::Spawn => {                                                                                                                                                                   
                if !accounts[0].executable || accounts[0].loader != Pubkey::default() {                                                                                                                     
                    Err(Error::AccountNotFinalized)?;                                                                                                                                                       
                }                                                                                                                                                                                           
                accounts[0].loader = accounts[0].owner;                                                                                                                                                     
                accounts[0].owner = tx.account_keys[0];                                                                                                                                                     
            }                                                                                                                                                                                               
```

**Comment by mvines** [comment]
Fixed by #1916
